lib/src/atom/matcher.rs

```diff
@@ -336,15 +336,18 @@ fn atom_contains_variable(atom: &Atom, var: &VariableAtom) -> bool {
 }
 
 pub fn apply_bindings_to_bindings(from: &Bindings, to: &Bindings) -> Result<Bindings, ()> {
+    log::trace!("apply_bindings_to_bindings: from: {}, to: {}", from, to);
     let mut res = Bindings::new();
     for (key, value) in to {
         let applied = apply_bindings_to_atom(value, from);
         // Check that variable is not expressed via itself, if so it is
         // a task for unification not for matching
         if !matches!(applied, Atom::Variable(_)) && atom_contains_variable(&applied, key) {
+            log::trace!("apply_bindings_to_bindings: self expression: key: {}, applied: {}", key, applied);
```

Comment:
better log message would be "rejecting binding, variable is expressed via itself"

Reply:
Fixed